docs/utilities/parser.md

```diff
@@ -2,168 +2,85 @@
-
-class OrderItem(BaseModel):
-	id: int
-	quantity: int
-	description: str
+Parser supports Pydantic v2. Each Pydantic version requires different dependencies before you can use Parser.
 
-class Order(BaseModel):
-	id: int
-	description: str
-	items: List[OrderItem] # nesting models are supported
-	optional_field: Optional[str] = None # this field may or may not be available when parsing
+```python
+pip install aws-lambda-powertools
 ```
 
-These are simply Python classes that inherit from BaseModel. **Parser** enforces type hints declared in your model at runtime.
+!!! info "This is not necessary if you're installing Powertools for AWS Lambda (Python) via [Lambda Layer/SAR](../index.md#lambda-layer){target="\_blank"}"
+
+???+ warning
+    This will increase the compressed package size by >10MB due to the Pydantic dependency.
 
-### Parsing events
+    To reduce the impact on the package size at the expense of 30%-50% of its performance[Pydantic can also be
+    installed without binary files](https://pydantic-docs.helpmanual.io/install/#performance-vs-package-size-trade-off){target="_blank" rel="nofollow"}:
 
-You can parse inbound events using **event_parser** decorator, or the standalone `parse` function. Both are also able to parse either dictionary or JSON string as an input.
+    Pip example:`SKIP_CYTHON=1 pip install --no-binary pydantic aws-lambda-powertools[parser]`
 
-#### event_parser decorator
 
-Use the decorator for fail fast scenarios where you want your Lambda function to raise an exception in the event of a malformed payload.
+You can also add as a dependency in your preferred tool: e.g., requirements.txt, pyproject.toml.
 
-`event_parser` decorator will throw a `ValidationError` if your event cannot be parsed according to the model.
+### Data Model with Parse
 
-???+ note
-    **This decorator will replace the `event` object with the parsed model if successful**. This means you might be careful when nesting other decorators that expect `event` to be a `dict`.
+Define models by inheriting from `BaseModel` to parse incoming events. Pydantic then validates the data, ensuring all fields adhere to specified types and guaranteeing data integrity.
 
-```python hl_lines="19" title="Parsing and validating upon invocation with event_parser decorator"
-from aws_lambda_powertools.utilities.parser import event_parser, BaseModel
-from aws_lambda_powertools.utilities.typing import LambdaContext
-from typing import List, Optional
+#### Event parser
 
-import json
+The `event_parser` decorator automatically parses and validates the event.
 
-class OrderItem(BaseModel):
-	id: int
-	quantity: int
-	description: str
 
-class Order(BaseModel):
-	id: int
-	description: str
-	items: List[OrderItem] # nesting models are supported
-	optional_field: Optional[str] = None # this field may or may not be available when parsing
+```python title="getting_started_with_parser.py" hl_lines="2 8"
+--8<-- "examples/parser/src/getting_started_with_parser.py"
+```
```

Comment:
Great example, but there is a problem here. When using `@event_parser` you can't catch the exception inside the Lambda because the validation [happens before](https://github.com/aws-powertools/powertools-lambda-python/blob/7eb6026264b434f0b1017b41b2621c165c2b5bde/aws_lambda_powertools/utilities/parser/parser.py#L110) to execute the Lambda handler. So, in this case you can remove the try/except block and add a new line before the `The @event_parser(model=MyEvent) automatically` explaining that this function fails fast.

Also, pelase create a new JSON file and add as example of the payload. Something like this: https://docs.powertools.aws.dev/lambda/python/latest/utilities/idempotency/#sample-event.

In this case your payload is just id and name:

{
"id": "bla",
"name": "bla"
}

Reply:
Thanks, created the example!